cypress/e2e/buy-shirt.cy.ts

```diff
@@ -1,26 +1,41 @@
+import {MenuContentPage, AddressStepPage, LoginPage, PaymentStepPage,
+  ProductsListPage, ShippingStepPage, ShoppingCartPage} from "../page/index";
+
+const menuContentPage = new MenuContentPage();
+const addressStepPage = new AddressStepPage();
+const loginPage = new LoginPage();
+const paymentStepPage = new PaymentStepPage();
+const productsListPage = new ProductsListPage();
+const shippingStepPage = new ShippingStepPage();
+const shoppingCartPage = new ShoppingCartPage();
+
 describe("Buy a t-shirt", () => {
   it("Then the t-shirt should be bought", () => {
-    cy.visit("http://automationpractice.com/");
-    cy.get("#block_top_menu > ul > li:nth-child(3) > a").click();
-    cy.get("#center_column a.button.ajax_add_to_cart_button.btn.btn-default").click();
-    cy.get(".button-container > .button-medium > span").click();
-    cy.get(".cart_navigation span").click();
+    // Visiting T-Shirt page
+    menuContentPage.visitMenuContentPage();
+    menuContentPage.goToTShirtMenu();
+
+    // products list
+    productsListPage.addToCart();
+    productsListPage.goToCheckout();
+
+    // Shopping cart
+    shoppingCartPage.goProceedToCheckout();
 
     // Login user
-    cy.get("#email").type("aperdomobo@gmail.com");
-    cy.get("#passwd").type("WorkshopProtractor");
-    cy.get("#SubmitLogin > span").click();
+    loginPage.login("aperdomobo@gmail.com", "WorkshopProtractor");
+    loginPage.signIn();
 
     // Address
-    cy.get(".cart_navigation > .button > span").click();
-    cy.get("#cgv").click();
+    addressStepPage.proceedToCheckout();
 
     // Shipping
-    cy.get(".cart_navigation > .button > span").click();
-    cy.get(".bankwire").click();
+    shippingStepPage.acceptTerms();
+    shippingStepPage.proceedToCheckout();
 
     // Payment
-    cy.get("#cart_navigation > .button > span").click();
-    cy.get(".cheque-indent > .dark").should("have.text", "Your order on My Store is complete.");
+    paymentStepPage.paymentMethod();
+    paymentStepPage.confirmMethod();
+    paymentStepPage.checkResultText();
```

Comment:
I suggest the following, the asserts are most common on the it methods, not inside the Page object, it would be better to return the element and assert there
`paymentStepPage.getOrderStatus().should('have.text',expectedMessage)`